src/vsr/replica.zig

```diff
@@ -6508,6 +6514,19 @@ pub fn ReplicaType(
                 return .busy;
             }
 
+            const view_headers_op_max = self.view_headers.array.get(0).op;
+            if (view_headers_op_max > self.op) {
+                assert(view_headers_op_max <= self.op_prepare_max());
+                self.op = view_headers_op_max;
+
+                var op = self.commit_max + 1;
+                while (op <= view_headers_op_max) : (op += 1) {
+                    const op_header =
+                        &self.view_headers.array.const_slice()[view_headers_op_max - op];
+                    self.replace_header(op_header);
+                }
+            }
```

Comment:
This block is worth a comment!

Reply:
https://github.com/tigerbeetle/tigerbeetle/pull/2236/commits/36fef729e334b6fefa0e6ad20fe7f68c0367b457#diff-e95e4dc6c1aae349e798cfeed22ad01f830cce0e654e6b4b765b3e7b0c677b6aR6522-R6524